de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/navigation/proxy/SelectFolderByPathLoadHandler.java

```diff
@@ -141,27 +152,25 @@ public void onLoad(LoadEvent<Folder, List<Folder>> event) {
     }
 
     /**
-     * Verify if the desired selected folder has already been loaded in the {@link DiskResourceView}.
+     * Verify if the desired selected folder has already been loaded in the {@link NavigationView}.
      * This only needs to occur once, but only after the root folders have been loaded. This method will
      * determine how much of the target folder's path has already been loaded, and if a parent of the
      * target folder needs to be expanded in order to trigger a load of its children, or a refresh to
      * force a reload of its children, in order to start the {@link #onLoad(LoadEvent)} callbacks.
+     *
+     * Must be called after this handler has been registered.
      */
-    private void initPathsToLoad() {
-        // Check if the requested folder's path is under a known root path.
-        boolean matched = false;
-        for (Folder root : navigationPresenter.getRootItems()) {
-            if (folderToSelect.getPath().startsWith(root.getPath())) {
-                matched = true;
-                break;
-            }
+    protected void initPathsToLoad() {
+        rootsLoaded = navigationPresenter.rootsLoaded();
+        if (!rootsLoaded) {
+            return;
         }
 
-        if (!matched) {
+        // Check if the requested folder's path is under a known root path.
+        if (!navigationPresenter.isPathUnderKnownRoot(folderToSelect.getPath())) {
```

Comment:
0% test coverage. This is using a new method from the navigation presenter. The tests should be expanded, or new test added to ensure appropriate coverage.